qmi/instruments/newport/smc_100cc.py

```diff
@@ -30,3 +44,542 @@ def __init__(self,
+
+        self._scpi_protocol.write(self._build_command(
+            "KI", integral_gain))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        if persist:
+            self.exit_configuration_state(controller_address)
+
+    @rpc_method
+    def get_proportional_gain(self, controller_address: Optional[int] = None) -> float:
+        """
+        Get the proportional gain of the PID control loop.
+
+        Parameters:
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised value of the controller address.
+
+        Returns:
+            The proportional gain of the PID control loop.
+        """
+        _logger.info(
+            "Getting proportional gain of the PID control loop of instrument [%s]", self._name)
+        # instrument must be in configuration state to get the proportional gain of the PID control loop value.
+        self.reset(controller_address)
+        self.enter_configuration_state(controller_address)
+        res = self._scpi_protocol.ask(
+            self._build_command("KP?"))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        self.exit_configuration_state(controller_address)
+        return float(res[3:])
+
+    @rpc_method
+    def set_proportional_gain(
+            self, proportional_gain: float, persist: bool = False, controller_address: Optional[int] = None) -> None:
+        """
+        Set the proportional gain of the PID control loop.
+
+        Parameters:
+            proportional_gain:    New proportional_gain value in Volt * second/preset unit.
+            persist:            Flag to indicate if the proportional gain should be persisted to the controller's memory,
+                                so it is still available after powering down the controller. When not persisted, the
+                                proportional gain is the one stored in the controller's memory.
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised proportional_gain of the controller address.
+        """
+        if 1E12 <= proportional_gain or proportional_gain < 0:
+            raise QMI_InstrumentException(
+                f"Provided value {proportional_gain} not in valid range 0 >= proportional_gain > 1E12.")
+
+        _logger.info(
+            "Setting proportional gain of the PID control loop of instrument [%s] to [%f]",
+            self._name, proportional_gain
+        )
+        # instrument must be in configuration state to persist the set proportional gain of the PID control loop.
+        self.controller_address = controller_address
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+        self._scpi_protocol.write(self._build_command(
```

Comment:
Same ass the `FD` command. It cannot be called in the Ready state. So you would have to make sure the controller is in the Disable state here